### ISSUE TITLE
Varying results over multiple runs

### ISSUE THREAD

**Issue by jbedorf** [terrencepreilly/darglint#83]
Hi,

I'm seeing different results when running `darglint` multiple times on the same file.

I tracked down the issue to the number of workers, since if I decrease it from 4 to 1 (https://github.com/terrencepreilly/darglint/blob/master/darglint/integrity_checker.py#L89 ) the problem does not occur.

I can't share the source code, but `darglint `seems to mix the docstrings of different functions, leading to something like this:

```
run1: No errors
run2: DAR202: Excess "Returns" in Docstring: + return
run2: DAR401: Missing exception(s) in Raises section: -r ValueError
run3...8: No Errors
run9: DAR402: Excess exception(s) in Raises section: +r ValueError
run10...19: No errors
run20: DAR102: Excess parameter(s) in Docstring: + state
```

It appears that the docstrings of multiple functions are mixed when evaluated. For example the final error the one that says 'Excess parameter' reports this parameter on a line number that falls inside `fooB()` while the 'state' is mentioned in the docstring of `fooA()`.

**Comment by terrencepreilly** [comment]
Could I have a little more information about how you invoked darglint?  What settings were you using, and what version are you running?

**Comment by jbedorf** [comment]
Sure, I used it in the following setting:
* Python 3.6.9
* darglint 1.2.2
* Default settings (Google style docstrings, strictness 'full')
  - Invoke via: `darglint  -v 2 -s google -z full`
* Files that report problems in general have 100+ lines of code.

**Comment by jbedorf** [comment]
I tested it a bit more. It appears that when [this line](https://github.com/terrencepreilly/darglint/blob/master/darglint/integrity_checker.py#L113) is accessed by multiple workers at the same time, the results of the parsing gets mixed. If I move the docstring parsing outside of the `run_checks` and into the `schedule` function the problem does not occur.
Tested by running darglint 1000 times on the same file:
- Original code: 15 out of the 1000 runs reported errors that were not there.
- Moved parsing location: 0 out of 1000 runs reported errors.
